posthog/hogql_queries/insights/retention_query_runner.py

```diff
@@ -0,0 +1,380 @@
+from datetime import datetime, timedelta
+from math import ceil
+from typing import Any, Dict
+from typing import Optional
+
+from posthog.caching.insights_api import BASE_MINIMUM_INSIGHT_REFRESH_INTERVAL, REDUCED_MINIMUM_INSIGHT_REFRESH_INTERVAL
+from posthog.caching.utils import is_stale
+from posthog.constants import (
+    PAGEVIEW_EVENT,
+    TREND_FILTER_TYPE_ACTIONS,
+    TREND_FILTER_TYPE_EVENTS,
+    TRENDS_LINEAR,
+    RetentionQueryType,
+)
+from posthog.hogql import ast
+from posthog.hogql.parser import parse_select, parse_expr
+from posthog.hogql.printer import to_printed_hogql
+from posthog.hogql.property import action_to_expr, property_to_expr
+from posthog.hogql.query import execute_hogql_query
+from posthog.hogql.timings import HogQLTimings
+from posthog.hogql_queries.query_runner import QueryRunner
+from posthog.hogql_queries.utils.query_date_range import QueryDateRangeWithIntervals
+from posthog.models import Entity, Team
+from posthog.models import RetentionFilter
+from posthog.models.action.util import Action
+from posthog.models.filters.mixins.retention import RetentionDateDerivedMixin
+from posthog.models.filters.mixins.utils import cached_property
+from posthog.models.team import WeekStartDay
+from posthog.queries.retention.types import CohortKey
+from posthog.queries.util import correct_result_for_sampling
+from posthog.queries.util import get_trunc_func_ch
+from posthog.schema import (
+    HogQLQueryModifiers,
+    RetentionQueryResponse,
+    IntervalType,
+)
+from posthog.schema import RetentionQuery, RetentionType
+
+
+class RetentionQueryRunner(QueryRunner):
+    query: RetentionQuery
+    query_type = RetentionQuery
+    EVENT_TABLE_ALIAS = "e"
+
+    def __init__(
+        self,
+        query: RetentionQuery | Dict[str, Any],
+        team: Team,
+        timings: Optional[HogQLTimings] = None,
+        modifiers: Optional[HogQLQueryModifiers] = None,
+        in_export_context: Optional[bool] = None,
+        event_query_type: Optional[RetentionQueryType] = None,
+    ):
+        super().__init__(query, team=team, timings=timings, modifiers=modifiers, in_export_context=in_export_context)
+        self.event_query_type = event_query_type
+        self.old_filter = RetentionFilter(data=self.query.retentionFilter.model_dump())
```

Comment:
We'll want to change the system such that we're not referencing the old filter classes/code anymore. Ideally we should/could get what we need out of `self.query.retentionFilter` directly.